schemas/dns.yml

```diff
@@ -0,0 +1,210 @@
+
+        For example, the registered domain for "foo.google.com" is "google.com".
+
+        This value can be determined precisely with a list like the public
+        suffix list (http://publicsuffix.org). Trying to approximate this by
+        simply taking the last two labels will not work well for TLDs such as "co.uk".
+      example: google.com
+
+    - name: question.length
+      level: extended
+      type: long
+      short: The length of the name being queried (in bytes).
+      description: >
+        The length of the name being queried (in bytes).
+
+        This is useful to detect data exfiltration via DNS queries.
+      example: 42
+
+    - name: answers_count
+      level: extended
+      type: long
+      description: >
+        The number of resource records contained in the `dns.answers` field.
+      example: 3
+
+    - name: answers
+      level: extended
+      type: object
+      short: Array of DNS answers.
+      description: >
+        An array containing an object for each answer section returned by
+        the server.
+
+        The main keys that should be present in these objects are defined by ECS.
+        Records that have more information may contain more keys than what
+        ECS defines.
+
+        Not all DNS data sources give all details about DNS answers.
+        At minimum, answer objects must contain the `data` key.
+        If more information is available, map as much of it to ECS as possible,
+        and add any additional fields to the answer objects as custom fields.
+
+    - name: answers.name
+      level: extended
+      type: keyword
+      short: The domain name to which this resource record pertains.
+      description: >
+        The domain name to which this resource record pertains.
+
+        If a chain of CNAME is being resolved, each answer's `name` should be
+        the one that corresponds with the answer's `data`. It should not simply
+        be the original `question.name` repeated.
+      example: www.google.com
+
+    - name: answers.type
+      level: extended
+      type: keyword
+      description: The type of data contained in this resource record.
+      example: CNAME
+
+    - name: answers.class
+      level: extended
+      type: keyword
+      description: The class of DNS data contained in this resource record.
+      example: IN
+
+    - name: answers.ttl
+      level: extended
+      type: long
+      description: >
+        The time interval in seconds that this resource record may be cached
+        before it should be discarded. Zero values mean that the data should
+        not be cached.
+      example: 180
+
+    - name: answers.data
+      level: extended
+      type: keyword
+      short: The data describing the resource.
+      description: >
+        The data describing the resource.
+
+        The meaning of this data depends on the type and class of the resource
+        record.
+      example: 10.10.10.10
+
+    - name: grouped.ip
```

Comment:
Instead of `dns.grouped.ip`, can we call this something else? I find it hard to immediately grasp what a field called `dns.grouped.ip` contains - could be the IPs of the requestor, could be the IPs for a reverse DNS request.

How about `dns.resolved_ip`?

Reply:
I like `dns.resolved_ip`, yes. It's more representative of the content of the field.